src/database/subClasses/GenomeMethods.java

```diff
@@ -560,8 +563,22 @@ public int removeChainFiles(String fromVersion, String toVersion)
 			throws SQLException {
 
 		int resCount = 0;
+		ChainFiles cf = null;
 
-		ChainFiles cf = getChainFiles(fromVersion, toVersion);
+		String getChainFiles =
+				"SELECT * FROM Chain_File NATURAL JOIN Chain_File_Files " +
+				"WHERE (FromVersion ~~* ?)" + " AND (ToVersion ~~* ?)";
+
+		PreparedStatement getChainFilesStmt = conn.prepareStatement(getChainFiles);
+		getChainFilesStmt.setString(1, fromVersion);
+		getChainFilesStmt.setString(2, toVersion);
+		ResultSet getChainFilesRs = getChainFilesStmt.executeQuery();
+
+		if (getChainFilesRs.next()) {
+			cf = new ChainFiles(getChainFilesRs);
+		}
```

Comment:
A tuple in `Chain_File` table is uniquely identified by `FromVersion` and `ToVersion` attributes. But in `Chain_File_Files`, the same attributes are used (which actually make a foreign reference to `Chain_File`) plus `fileName` attribute. Therefore a such query might return several tuples, since the where clause specifies only two of the three identifiers. I see you have only a single `next()` statement on the result set, are you sure that's how it's supposed to be? That only one file is handled and the rest are ignored?